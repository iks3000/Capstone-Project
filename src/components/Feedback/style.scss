@import '../../assets/styles/_variables.scss';
@import '../../assets/styles/_extends.scss';


.feedback {
    margin: 50px 0 0;

    &__title {
        @extend %title;
    }

    &__container {
        margin-bottom: 40px;
        &:last-child {
            margin-bottom: 0;
        }
    }
    .info {
        padding: 15px;
        background-color: $gallery;
        margin-bottom: 20px;
    }

    &__user-block {
        display: flex;
        align-items: center;
        font: 400 12px/14px $primary-fonts;
    }

    &__user-img {
        border-radius: 50%;
        width: 50px;
        height: 50px;
        margin-right: 20px;
    }

    &__user-name {
        color: $oxfordBlue;
    }

    &__user-link {
        color: $green;
    }
}