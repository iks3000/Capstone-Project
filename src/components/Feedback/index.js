import { useState } from "react";
import Info from "../../components/Info";
import "./style.scss";

const Feedback = ({ data }) => {
  const [dataSeed] = useState(data);
  return (
    <div className="feedback" id="feedbackComponent">
      <h2 className="feedback__title">Feedbacks</h2>
      {dataSeed.map((dataFeedback, index) => (
        <div className="feedback__container" key={index}>
          <Info text={dataFeedback.feedback} />
          <div className="feedback__user-block">
            <img className="feedback__user-img" src={dataFeedback.reporter.photoUrl} alt="reporter icon" />
            <span className="feedback__user-name">{dataFeedback.reporter.name}</span>,&nbsp;
            <a className="feedback__user-link" href={dataFeedback.reporter.citeUrl} target="_blank" rel="noreferrer" >
              {dataFeedback.reporter.citeUrl.replace( /^(?:https?:\/\/)?(?:www\.)?/i, "")}
            </a>
          </div>
        </div>
      ))}
    </div>
  );
};

export default Feedback;
