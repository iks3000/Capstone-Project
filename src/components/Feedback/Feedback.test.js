import React from "react";
import { shallow } from "enzyme";
import Feedback from "./index";
import User1 from "../../assets/images/users/user1.png";
import User2 from "../../assets/images/users/user2.jpg";

it("should render Feedback component", () => {
  const mockItem = {
    data: [
      {
        feedback: "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus.",
        reporter: {
          photoUrl: <User1></User1>,
          name: "John Doe",
          citeUrl: "https://www.citeexample.com",
        },
      },
      {
        feedback: "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque.",
        reporter: {
          photoUrl: User2,
          name: "Smith Anderson",
          citeUrl: "https://www.citeexample.com",
        },
      },
    ],
  };
  expect(shallow(<Feedback {...mockItem} />)).toMatchSnapshot();
});
