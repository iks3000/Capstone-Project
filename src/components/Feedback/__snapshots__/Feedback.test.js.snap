// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Feedback component 1`] = `
<div
  className="feedback"
  id="feedbackComponent"
>
  <h2
    className="feedback__title"
  >
    Feedbacks
  </h2>
  <div
    className="feedback__container"
    key="0"
  >
    <Info
      text="Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus."
    />
    <div
      className="feedback__user-block"
    >
      <img
        alt="reporter icon"
        className="feedback__user-img"
        src={<user1.png />}
      />
      <span
        className="feedback__user-name"
      >
        John Doe
      </span>
      , 
      <a
        className="feedback__user-link"
        href="https://www.citeexample.com"
        rel="noreferrer"
        target="_blank"
      >
        citeexample.com
      </a>
    </div>
  </div>
  <div
    className="feedback__container"
    key="1"
  >
    <Info
      text="Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque."
    />
    <div
      className="feedback__user-block"
    >
      <img
        alt="reporter icon"
        className="feedback__user-img"
        src="user2.jpg"
      />
      <span
        className="feedback__user-name"
      >
        Smith Anderson
      </span>
      , 
      <a
        className="feedback__user-link"
        href="https://www.citeexample.com"
        rel="noreferrer"
        target="_blank"
      >
        citeexample.com
      </a>
    </div>
  </div>
</div>
`;
